### PR TITLE
Fp8 group gemm & sigmoid topk for dpsk v3

### PR DESCRIPTION
**Pull request by ZephyrZhao247**
### Description

Add FP8 E4M3 block-quantized group GEMM kernel for MoE layers, targeting DeepSeek V3/R1 inference on Blackwell B200 GPUs.

**Key features:**
- SM100 block-scaled FP8 UMMA (tcgen05.mma.kind::mxf8f6f4.block_scale)
- Per-128-K float32 scales converted to UE8M0 in-kernel
- 2D CTA work distribution: grid_dim.x for expert parallelism, grid_dim.y for N-dimension splitting
- Dedicated scale warp (warp 6) for parallel float32->UE8M0->transpose pipeline
- TMA for weight loading, 16-byte cp.async for routed input with SWIZZLE_128B
- TMA descriptor prefetching, scale barrier lookahead peeks

**Scale factor format:**
- Input: float32 per-row per-128-K-block (e.g., [256, 4096, 56] for DSV3 W13)
- Converted in-kernel to UE8M0 (8-bit exponent-only), packed into uint32 for UTCCP->TMEM

### Benchmark Results (B200, DeepSeek V3: N=4096, K=7168, 256 experts, top-8)

Latency comparison (median, microseconds):
| M | MPK-FP8 | MPK-BF16 | FI-grpGEMM | PyTorch |
|---|---------|----------|------------|---------|
| 1 | 96 | 94 | 71 | 146 |
| 2 | 174 | 160 | 114 | 270 |
| 4 | 328 | 282 | 204 | 592 |
| 8 | 565 | 494 | 374 | 1246 |
| 16 | 1037 | 911 | 743 | 2579 |

- MPK-FP8 is 1.13x of MPK-BF16 (13% overhead from scale handling)
- MPK-FP8 is 2.5x faster than PyTorch BF16 per-expert matmul

FP8 grid config sweep (128 CTAs total):
| M | (4,32) | (8,16) | (16,8) | (32,4) |
|---|--------|--------|--------|--------|
| 1 | 96 | 96 | 156 | 311 |
| 8 | 577 | 565 | 629 | 623 |
| 16 | 1055 | 1032 | 1100 | 1245 |

Best config: (8, 16, 1) -- 8 expert CTAs x 16 N-splits

### Test Plan

- [x] 39 correctness tests pass (single-CTA, 2D grid 8x16, varying n_splits 1-32, 10 random seeds)
- [x] Benchmark with DeepSeek V3 dimensions (N=4096, K=7168, 256 experts, top-8)
- [x] Compare against BF16 kernel, FlashInfer group_gemm FP8, torch.bmm BF16

## Sigmoid topk for dpsk v3

This task is tailored to the routing logic ok Deepseek v3.

batch_size | sigmoid | softmax | ratio
-- | -- | -- | --
1 | 14.5 us | 11.2 us | 1.29x
2 | 15.4 us | 11.6 us | 1.32x
4 | 14.7 us | 11.8 us | 1.25x
8 | 18.2 us | 14.7 us | 1.24x

### CODE REVIEW

File: include/mirage/persistent_kernel/tasks/blackwell/fp8_group_gemm_sm100.cuh [hunk 1]
```diff
@@ -0,0 +1,1725 @@
...
+//
+//   swizzled_byte_offset = linear_offset ^ (((linear_offset >> 7) & 7) << 4)
+//
+// This XOR-based swizzle eliminates bank conflicts for 128-byte aligned
+// access patterns. The UMMA descriptor encodes this as
+// LayoutType::SWIZZLE_128B.
+//
+// A (weight): loaded via TMA which handles swizzling automatically.
+// B (input):  loaded via manual cp.async with explicit swizzle computation
+//             in the DMA warp (since we need per-token routing checks).
```
Comment (xinhaoc) [inline include/mirage/persistent_kernel/tasks/blackwell/fp8_group_gemm_sm100.cuh:127]: let's remove some of the comments and add licence.

**Comment by xinhaoc** [comment]
Overall Looks great, thanks for the PR!  
Just don't forget to remove the long comment and add license to our runtime and kernel files.

File: include/mirage/persistent_kernel/tasks/blackwell/fp8_group_gemm_sm100.cuh (hunk 1 above)
Comment (ZephyrZhao247) [reply include/mirage/persistent_kernel/tasks/blackwell/fp8_group_gemm_sm100.cuh:127]: Done.
I preserved some comments that are helpful to human readers.

**Review by xinhaoc — APPROVED** [review]
